gems/smithy/lib/smithy/anvil/client/views/shapes.rb

```diff
@@ -0,0 +1,241 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Anvil
+    module Client
+      module Views
+        # @api private
+        class Shapes < View
+          def initialize(plan)
+            @plan = plan
+            @model = plan.model
+            @service_shape = assemble_service_shape
+            @shapes, @operation_shapes = assemble_shapes
+            super()
+          end
+
+          def namespace
+            Tools::Namespace.namespace_from_gem_name(@plan.options[:gem_name])
+          end
+
+          def shapes_with_members
+            @shapes.reject { |s| s.members.empty? }
+          end
+
+          private
+
+          def assemble_service_shape
+            shape = Vise::ServiceIndex.new(@model).service
+            shape_data = shape.values.first
+
+            ServiceShape.new(
+              id: shape.keys.first,
+              traits: filter_traits(shape_data['traits']),
+              version: shape_data['version']
+            )
+          end
+
+          def assemble_shapes
+            serializable_shapes = []
+            operation_shapes = []
+
+            @model['shapes'].each do |id, shape|
```

Comment:
This is where we need a service walker or some sort of "index" - the model may have shapes that are not part of the service's closure (ie, the graph of shapes connected to the service through its operations).

Reply:
Yes. I will work on adding this soon.